blocks/library/embed/index.js

```diff
@@ -19,17 +19,21 @@ registerBlock( 'core/embed', {
 		caption: children( 'figcaption' )
 	},
 
-	edit( { attributes, isSelected, setAttributes } ) {
+	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, title, caption } = attributes;
 
 		return (
 			<figure>
-				<iframe src={ url } title={ title } />
-				{ caption || isSelected ? (
+				<div className="iframe-overlay">
+					<iframe src={ url } title={ title } />
```

Comment:
To your last comment, this overlay is now just to prevent clicks going through iframes, so I named it like that. Not sure if it's intended for wider use.